Fix deprecation of Extract/Inject in otelgrpc (#7689)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- `WithSpanOptions` in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is deprecated.
  It is only used by the deprecated interceptor, and is unused by `NewClientHandler` and `NewServerHandler`. (#7601)

> - `Extract` and `Inject` in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is deprecated.
>   These functions were earlier exposed as public APIs, but are now considered unnecessary. (#7689)

Reviewer: ```suggestion
- `Extract` and `Inject` in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` are deprecated.
  These functions were initially exposed in the public API, but are now considered unnecessary. (#7689)
```
